crates/compression/src/eviction_policy.rs

```diff
@@ -0,0 +1,41 @@
+use std::collections::HashSet;
+
+use fuel_core_types::fuel_compression::RegistryKey;
+
+use crate::{
+    ports::EvictorDb,
+    tables::{
+        PerRegistryKeyspace,
+        RegistryKeyspace,
+    },
+};
+
+pub struct CacheEvictor {
+    /// Set of keys that must not be evicted
+    pub keep_keys: PerRegistryKeyspace<HashSet<RegistryKey>>,
+}
+
+impl CacheEvictor {
+    /// Get a key, evicting an old value if necessary
+    pub fn next_key<D>(
+        &mut self,
+        db: &mut D,
+        keyspace: RegistryKeyspace,
+    ) -> anyhow::Result<RegistryKey>
+    where
+        D: EvictorDb,
+    {
+        // Pick first key not in the set
+        // TODO: use a proper algo, maybe LRU?
+        let mut key = db.read_latest(keyspace)?;
+        while self.keep_keys[keyspace].contains(&key) {
+            key = key.next();
+            assert_ne!(key, RegistryKey::ZERO, "Ran out of keys");
```

Comment:
It seems we can hit this assert just during iteration. 

<img width="790" alt="image" src="https://github.com/user-attachments/assets/6189fa2d-f5d9-45bb-992e-a292ae4b7560">

If the key is `Self::DEFAULT_VALUE - 1`, then the next key return a zero key.

Reply:
Yep it was clearly incorrect. Fixed in e0ddd85622.